src/Resource.lua

```diff
@@ -39,11 +38,39 @@ local Resource = {
 
 function Resource:take(name, pos)
     LOG.DEBUG("name:" .. name)
-    return self.supplies[name].takeObject({
-        position = pos,
-        rotation = {0, 180, 0},
-        smooth = true
-    })
+
+    -- perform a raycast to find the topmost resource on the supply tile
+    local supply_tile = getObjectFromGUID(self.supply_tiles[name:lower()])
+    local hits = Physics.cast(
+        {
+            origin = supply_tile.getPosition() + Vector(0, -1, 0),
+            direction = Vector(0,1,0),
+            type = 1
+        }
+    )
+
+    -- reverse iterate to get the top-most resource first
+    local result = nil
+    for i = #hits, 1, -1 do
+        if hits[i].hit_object.getName():lower() == name:lower() and not hits[i].hit_object.isSmoothMoving() then
+            result = hits[i].hit_object
+            break
+        end
+    end
+
+    if result.getQuantity() ~= -1 then
```

Comment:
Will this result in an error when trying to find the getQuantity function if result is nil? Consider adding an early return if the result is nil.